apps/portal/prisma/new-schema.prisma

```diff
@@ -0,0 +1,229 @@
+
+model RolesOnUser {
+    id        String   @id @default(auto()) @map("_id") @db.ObjectId
+    role      Role     @relation(fields: [roleId], references: [id])
+    roleId    String
+    user      User     @relation(fields: [userId], references: [id])
+    userId    String
+    createdAt DateTime @default(now())
+
+    @@unique([userId, roleId])
+}
+
+model User {
+    id            String        @id @default(auto()) @map("_id") @db.ObjectId
+    name          String?
+    email         String?       @unique
+    emailVerified DateTime?
+    image         String?
+    accounts      Account[]
+    sessions      Session[]
+    profile       Profile?
+    roles         RolesOnUser[]
+
+    applications    UserAppJunction[]   @relation("UserLink")
+
+    @@map("users")
+}
+
+model VerificationToken {
+    id         String   @id @default(auto()) @map("_id") @db.ObjectId
+    identifier String
+    token      String   @unique
+    expires    DateTime
+
+    @@unique([identifier, token])
+}
+
+model Profile {
+    id               String    @id @default(auto()) @map("_id") @db.ObjectId
+    firstName        String
+    lastName         String
+    email            String    @unique
+    netid            String    @unique
+    classStanding    String
+    major            String
+    utdStudent       Boolean
+    roles            String[]
+    membershipStatus Boolean
+    membershipTS     DateTime?
+    resume           Boolean
+    resumeTS         DateTime?
+
+    user User @relation(fields: [userId], references: [id])
+
+    userId String             @unique @db.ObjectId
+    events EventReservation[]
+
+    officerId   String             @unique @db.ObjectId
+    officer     Officer            @relation(fields: [officerId], references: [id])
+}
+
+model Event {
+    id          String             @id @default(auto()) @map("_id") @db.ObjectId
+    summary     String
+    description String
+    url         String
+    location    String
+    start       DateTime
+    end         DateTime
+    profiles    EventReservation[]
+
+    @@map("events")
+}
+
+model TypeformApplication {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    active              Boolean
+    description         String
+    endpoint            String
+    externalResourceUrl String
+    typeformId          String
+    typeformName        String
+
+    @@map("typeform_applications")
+}
+
+model Application {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    createdAt           DateTime
+    expireDate          DateTime
+    type                String?
+    role                String?
+
+    users               UserAppJunction[]   @relation("AppLink")
+    departments         UserAppJunction[]   @relation("AppLink")
+
+    @@map("applications")
+}
+
+model Department {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    deptName            String
+
+    officers            Officer[]
```

Comment:
Based on [this part]() of the documentation, would we do something like this for officer-department? 
> officerIDs String[]
    officers    Officer[] @relation(fields: [officerIDs], references: [id])

Reply:
yeah so in prisma, there are also implicit m-to-n relations where you just put it like that so instead of having a junction table, prisma takes care of building that beneath it.

that's why beneath it, i provided a junction table in case it is needed but the connection between officers and departments is a simple m-to-n so the implicit case should work here

Reply:
you are correct - was looking at the wrong thing